// @openrouter/sdk/esm/models/operations/listbyokkeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 953de50b8d0e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
import * as models from "../index.js";
/**
 * Optional provider slug to filter by (e.g. `openai`, `anthropic`, `amazon-bedrock`).
 */
export const Provider = {
    Ai21: "ai21",
    AionLabs: "aion-labs",
    Akashml: "akashml",
    Alibaba: "alibaba",
    AmazonBedrock: "amazon-bedrock",
    AmazonNova: "amazon-nova",
    Ambient: "ambient",
    Anthropic: "anthropic",
    ArceeAi: "arcee-ai",
    AtlasCloud: "atlas-cloud",
    Avian: "avian",
    Azure: "azure",
    Baidu: "baidu",
    Baseten: "baseten",
    BlackForestLabs: "black-forest-labs",
    Byteplus: "byteplus",
    Cerebras: "cerebras",
    Chutes: "chutes",
    Cirrascale: "cirrascale",
    Clarifai: "clarifai",
    Cloudflare: "cloudflare",
    Cohere: "cohere",
    Crusoe: "crusoe",
    Darkbloom: "darkbloom",
    Deepinfra: "deepinfra",
    Deepseek: "deepseek",
    Dekallm: "dekallm",
    Digitalocean: "digitalocean",
    Featherless: "featherless",
    Fireworks: "fireworks",
    Friendli: "friendli",
    Gmicloud: "gmicloud",
    GoogleAiStudio: "google-ai-studio",
    GoogleVertex: "google-vertex",
    Groq: "groq",
    Hyperbolic: "hyperbolic",
    Inception: "inception",
    Inceptron: "inceptron",
    InferenceNet: "inference-net",
    Infermatic: "infermatic",
    Inflection: "inflection",
    IoNet: "io-net",
    Ionstream: "ionstream",
    Liquid: "liquid",
    Mancer: "mancer",
    Mara: "mara",
    Minimax: "minimax",
    Mistral: "mistral",
    Modelrun: "modelrun",
    Modular: "modular",
    Moonshotai: "moonshotai",
    Morph: "morph",
    Ncompass: "ncompass",
    Nebius: "nebius",
    NexAgi: "nex-agi",
    Nextbit: "nextbit",
    Novita: "novita",
    Nvidia: "nvidia",
    OpenInference: "open-inference",
    Openai: "openai",
    Parasail: "parasail",
    Perceptron: "perceptron",
    Perplexity: "perplexity",
    Phala: "phala",
    Poolside: "poolside",
    Recraft: "recraft",
    Reka: "reka",
    Relace: "relace",
    Sambanova: "sambanova",
    Seed: "seed",
    Siliconflow: "siliconflow",
    Sourceful: "sourceful",
    Stepfun: "stepfun",
    Streamlake: "streamlake",
    Switchpoint: "switchpoint",
    Together: "together",
    Upstage: "upstage",
    Venice: "venice",
    Wandb: "wandb",
    Xai: "xai",
    Xiaomi: "xiaomi",
    ZAi: "z-ai",
};
/** @internal */
export const Provider$outboundSchema = openEnums
    .outboundSchema(Provider);
/** @internal */
export const ListBYOKKeysRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int()).optional(),
    limit: z.int().optional(),
    workspaceId: z.string().optional(),
    provider: Provider$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        workspaceId: "workspace_id",
    });
});
export function listBYOKKeysRequestToJSON(listBYOKKeysRequest) {
    return JSON.stringify(ListBYOKKeysRequest$outboundSchema.parse(listBYOKKeysRequest));
}
/** @internal */
export const ListBYOKKeysResponse$inboundSchema = z.object({
    Result: models.ListBYOKKeysResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listBYOKKeysResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListBYOKKeysResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListBYOKKeysResponse' from JSON`);
}
