// @openrouter/sdk/esm/models/anthropictextblockparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 81e2e56dbe5c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicCitationCharLocationParam$outboundSchema, } from "./anthropiccitationcharlocationparam.js";
import { AnthropicCitationContentBlockLocationParam$outboundSchema, } from "./anthropiccitationcontentblocklocationparam.js";
import { AnthropicCitationPageLocationParam$outboundSchema, } from "./anthropiccitationpagelocationparam.js";
import { AnthropicCitationSearchResultLocation$outboundSchema, } from "./anthropiccitationsearchresultlocation.js";
import { AnthropicCitationWebSearchResultLocation$outboundSchema, } from "./anthropiccitationwebsearchresultlocation.js";
/** @internal */
export const Citation$outboundSchema = z
    .union([
    AnthropicCitationCharLocationParam$outboundSchema,
    AnthropicCitationContentBlockLocationParam$outboundSchema,
    AnthropicCitationPageLocationParam$outboundSchema,
    AnthropicCitationSearchResultLocation$outboundSchema,
    AnthropicCitationWebSearchResultLocation$outboundSchema,
]);
export function citationToJSON(citation) {
    return JSON.stringify(Citation$outboundSchema.parse(citation));
}
/** @internal */
export const AnthropicTextBlockParam$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    citations: z.nullable(z.array(z.union([
        AnthropicCitationCharLocationParam$outboundSchema,
        AnthropicCitationContentBlockLocationParam$outboundSchema,
        AnthropicCitationPageLocationParam$outboundSchema,
        AnthropicCitationSearchResultLocation$outboundSchema,
        AnthropicCitationWebSearchResultLocation$outboundSchema,
    ]))).optional(),
    text: z.string(),
    type: z.literal("text"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function anthropicTextBlockParamToJSON(anthropicTextBlockParam) {
    return JSON.stringify(AnthropicTextBlockParam$outboundSchema.parse(anthropicTextBlockParam));
}
