// @openrouter/sdk/esm/models/byokproviderslug.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f8e7a5a4b0a2
 */
import * as openEnums from "../types/enums.js";
/**
 * The upstream provider this credential authenticates against, as a lowercase slug (e.g. `openai`, `anthropic`, `amazon-bedrock`).
 */
export const BYOKProviderSlug = {
    Ai21: "ai21",
    AionLabs: "aion-labs",
    Akashml: "akashml",
    Alibaba: "alibaba",
    AmazonBedrock: "amazon-bedrock",
    AmazonNova: "amazon-nova",
    Ambient: "ambient",
    Anthropic: "anthropic",
    ArceeAi: "arcee-ai",
    AtlasCloud: "atlas-cloud",
    Avian: "avian",
    Azure: "azure",
    Baidu: "baidu",
    Baseten: "baseten",
    BlackForestLabs: "black-forest-labs",
    Byteplus: "byteplus",
    Cerebras: "cerebras",
    Chutes: "chutes",
    Cirrascale: "cirrascale",
    Clarifai: "clarifai",
    Cloudflare: "cloudflare",
    Cohere: "cohere",
    Crusoe: "crusoe",
    Darkbloom: "darkbloom",
    Deepinfra: "deepinfra",
    Deepseek: "deepseek",
    Dekallm: "dekallm",
    Digitalocean: "digitalocean",
    Featherless: "featherless",
    Fireworks: "fireworks",
    Friendli: "friendli",
    Gmicloud: "gmicloud",
    GoogleAiStudio: "google-ai-studio",
    GoogleVertex: "google-vertex",
    Groq: "groq",
    Hyperbolic: "hyperbolic",
    Inception: "inception",
    Inceptron: "inceptron",
    InferenceNet: "inference-net",
    Infermatic: "infermatic",
    Inflection: "inflection",
    IoNet: "io-net",
    Ionstream: "ionstream",
    Liquid: "liquid",
    Mancer: "mancer",
    Mara: "mara",
    Minimax: "minimax",
    Mistral: "mistral",
    Modelrun: "modelrun",
    Modular: "modular",
    Moonshotai: "moonshotai",
    Morph: "morph",
    Ncompass: "ncompass",
    Nebius: "nebius",
    NexAgi: "nex-agi",
    Nextbit: "nextbit",
    Novita: "novita",
    Nvidia: "nvidia",
    OpenInference: "open-inference",
    Openai: "openai",
    Parasail: "parasail",
    Perceptron: "perceptron",
    Perplexity: "perplexity",
    Phala: "phala",
    Poolside: "poolside",
    Recraft: "recraft",
    Reka: "reka",
    Relace: "relace",
    Sambanova: "sambanova",
    Seed: "seed",
    Siliconflow: "siliconflow",
    Sourceful: "sourceful",
    Stepfun: "stepfun",
    Streamlake: "streamlake",
    Switchpoint: "switchpoint",
    Together: "together",
    Upstage: "upstage",
    Venice: "venice",
    Wandb: "wandb",
    Xai: "xai",
    Xiaomi: "xiaomi",
    ZAi: "z-ai",
};
/** @internal */
export const BYOKProviderSlug$inboundSchema = openEnums.inboundSchema(BYOKProviderSlug);
/** @internal */
export const BYOKProviderSlug$outboundSchema = openEnums.outboundSchema(BYOKProviderSlug);
