// @openrouter/sdk/esm/models/fusionservertoolconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 500c34bd192b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * Reasoning effort level for panelist and judge inner calls.
 */
export const FusionServerToolConfigEffort = {
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/** @internal */
export const FusionServerToolConfigEffort$outboundSchema = openEnums.outboundSchema(FusionServerToolConfigEffort);
/** @internal */
export const FusionServerToolConfigReasoning$outboundSchema = z.object({
    effort: FusionServerToolConfigEffort$outboundSchema.optional(),
    maxTokens: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        maxTokens: "max_tokens",
    });
});
export function fusionServerToolConfigReasoningToJSON(fusionServerToolConfigReasoning) {
    return JSON.stringify(FusionServerToolConfigReasoning$outboundSchema.parse(fusionServerToolConfigReasoning));
}
/** @internal */
export const FusionServerToolConfig$outboundSchema = z.object({
    analysisModels: z.array(z.string()).optional(),
    maxCompletionTokens: z.int().optional(),
    maxToolCalls: z.int().optional(),
    model: z.string().optional(),
    reasoning: z.lazy(() => FusionServerToolConfigReasoning$outboundSchema)
        .optional(),
    temperature: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        analysisModels: "analysis_models",
        maxCompletionTokens: "max_completion_tokens",
        maxToolCalls: "max_tool_calls",
    });
});
export function fusionServerToolConfigToJSON(fusionServerToolConfig) {
    return JSON.stringify(FusionServerToolConfig$outboundSchema.parse(fusionServerToolConfig));
}
