// @openrouter/sdk/esm/models/observabilityfilterrulesconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 297eacee53a2
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Logic = {
    And: "and",
    Or: "or",
};
export const Field = {
    Model: "model",
    Provider: "provider",
    SessionId: "session_id",
    UserId: "user_id",
    ApiKeyName: "api_key_name",
    FinishReason: "finish_reason",
    Input: "input",
    Output: "output",
    TotalCost: "total_cost",
    TotalTokens: "total_tokens",
    PromptTokens: "prompt_tokens",
    CompletionTokens: "completion_tokens",
};
export const Operator = {
    Equals: "equals",
    NotEquals: "not_equals",
    Contains: "contains",
    NotContains: "not_contains",
    Regex: "regex",
    StartsWith: "starts_with",
    EndsWith: "ends_with",
    Gt: "gt",
    Lt: "lt",
    Gte: "gte",
    Lte: "lte",
    Exists: "exists",
    NotExists: "not_exists",
};
/** @internal */
export const Logic$inboundSchema = openEnums
    .inboundSchema(Logic);
/** @internal */
export const Logic$outboundSchema = openEnums
    .outboundSchema(Logic);
/** @internal */
export const Field$inboundSchema = openEnums
    .inboundSchema(Field);
/** @internal */
export const Field$outboundSchema = openEnums
    .outboundSchema(Field);
/** @internal */
export const Operator$inboundSchema = openEnums
    .inboundSchema(Operator);
/** @internal */
export const Operator$outboundSchema = openEnums
    .outboundSchema(Operator);
/** @internal */
export const ObservabilityFilterRulesConfigValue$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const ObservabilityFilterRulesConfigValue$outboundSchema = z.union([z.string(), z.number()]);
export function observabilityFilterRulesConfigValueToJSON(observabilityFilterRulesConfigValue) {
    return JSON.stringify(ObservabilityFilterRulesConfigValue$outboundSchema.parse(observabilityFilterRulesConfigValue));
}
export function observabilityFilterRulesConfigValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityFilterRulesConfigValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityFilterRulesConfigValue' from JSON`);
}
/** @internal */
export const Rule$inboundSchema = z.object({
    field: Field$inboundSchema,
    operator: Operator$inboundSchema,
    value: z.union([z.string(), z.number()]).optional(),
});
/** @internal */
export const Rule$outboundSchema = z.object({
    field: Field$outboundSchema,
    operator: Operator$outboundSchema,
    value: z.union([z.string(), z.number()]).optional(),
});
export function ruleToJSON(rule) {
    return JSON.stringify(Rule$outboundSchema.parse(rule));
}
export function ruleFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rule$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rule' from JSON`);
}
/** @internal */
export const Group$inboundSchema = z.object({
    logic: Logic$inboundSchema.default("and"),
    rules: z.array(z.lazy(() => Rule$inboundSchema)),
});
/** @internal */
export const Group$outboundSchema = z.object({
    logic: Logic$outboundSchema.default("and"),
    rules: z.array(z.lazy(() => Rule$outboundSchema)),
});
export function groupToJSON(group) {
    return JSON.stringify(Group$outboundSchema.parse(group));
}
export function groupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Group$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Group' from JSON`);
}
/** @internal */
export const ObservabilityFilterRulesConfig$inboundSchema = z.object({
    enabled: z.boolean().default(true),
    groups: z.array(z.lazy(() => Group$inboundSchema)),
});
/** @internal */
export const ObservabilityFilterRulesConfig$outboundSchema = z.object({
    enabled: z.boolean().default(true),
    groups: z.array(z.lazy(() => Group$outboundSchema)),
});
export function observabilityFilterRulesConfigToJSON(observabilityFilterRulesConfig) {
    return JSON.stringify(ObservabilityFilterRulesConfig$outboundSchema.parse(observabilityFilterRulesConfig));
}
export function observabilityFilterRulesConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityFilterRulesConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityFilterRulesConfig' from JSON`);
}
