// @openrouter/sdk/esm/models/outputfusionservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 30bcfe965217
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, } from "./toolcallstatus.js";
/** @internal */
export const Stance$inboundSchema = z.object({
    model: z.string(),
    stance: z.string(),
});
export function stanceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Stance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Stance' from JSON`);
}
/** @internal */
export const Contradiction$inboundSchema = z
    .object({
    stances: z.array(z.lazy(() => Stance$inboundSchema)),
    topic: z.string(),
});
export function contradictionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Contradiction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Contradiction' from JSON`);
}
/** @internal */
export const PartialCoverage$inboundSchema = z.object({
    models: z.array(z.string()),
    point: z.string(),
});
export function partialCoverageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PartialCoverage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PartialCoverage' from JSON`);
}
/** @internal */
export const UniqueInsight$inboundSchema = z
    .object({
    insight: z.string(),
    model: z.string(),
});
export function uniqueInsightFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UniqueInsight$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UniqueInsight' from JSON`);
}
/** @internal */
export const Analysis$inboundSchema = z.object({
    blind_spots: z.array(z.string()),
    consensus: z.array(z.string()),
    contradictions: z.array(z.lazy(() => Contradiction$inboundSchema)),
    partial_coverage: z.array(z.lazy(() => PartialCoverage$inboundSchema)),
    unique_insights: z.array(z.lazy(() => UniqueInsight$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "blind_spots": "blindSpots",
        "partial_coverage": "partialCoverage",
        "unique_insights": "uniqueInsights",
    });
});
export function analysisFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Analysis$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Analysis' from JSON`);
}
/** @internal */
export const FailedModel$inboundSchema = z
    .object({
    error: z.string(),
    model: z.string(),
    status_code: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        "status_code": "statusCode",
    });
});
export function failedModelFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FailedModel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FailedModel' from JSON`);
}
/** @internal */
export const ResponseT$inboundSchema = z.object({
    model: z.string(),
});
export function responseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseT' from JSON`);
}
/** @internal */
export const OutputFusionServerToolItem$inboundSchema = z.object({
    analysis: z.lazy(() => Analysis$inboundSchema).optional(),
    error: z.string().optional(),
    failed_models: z.array(z.lazy(() => FailedModel$inboundSchema)).optional(),
    failure_reason: z.string().optional(),
    id: z.string().optional(),
    responses: z.array(z.lazy(() => ResponseT$inboundSchema)).optional(),
    status: ToolCallStatus$inboundSchema,
    type: z.literal("openrouter:fusion"),
}).transform((v) => {
    return remap$(v, {
        "failed_models": "failedModels",
        "failure_reason": "failureReason",
    });
});
export function outputFusionServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputFusionServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputFusionServerToolItem' from JSON`);
}
