// @openrouter/sdk/esm/models/presetwithdesignatedversion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6ebecbfabdd3
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { PresetDesignatedVersion$inboundSchema, } from "./presetdesignatedversion.js";
export const PresetWithDesignatedVersionStatus = {
    Active: "active",
    Disabled: "disabled",
    Archived: "archived",
};
/** @internal */
export const PresetWithDesignatedVersionStatus$inboundSchema = openEnums.inboundSchema(PresetWithDesignatedVersionStatus);
/** @internal */
export const PresetWithDesignatedVersion$inboundSchema = z.object({
    created_at: z.string(),
    creator_user_id: z.nullable(z.string()),
    description: z.nullable(z.string()),
    designated_version: z.nullable(PresetDesignatedVersion$inboundSchema),
    designated_version_id: z.nullable(z.string()),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    status: PresetWithDesignatedVersionStatus$inboundSchema,
    status_updated_at: z.nullable(z.string()),
    updated_at: z.string(),
    workspace_id: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "creator_user_id": "creatorUserId",
        "designated_version": "designatedVersion",
        "designated_version_id": "designatedVersionId",
        "status_updated_at": "statusUpdatedAt",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function presetWithDesignatedVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PresetWithDesignatedVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PresetWithDesignatedVersion' from JSON`);
}
